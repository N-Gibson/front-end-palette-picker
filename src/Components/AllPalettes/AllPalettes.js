import React from 'react';
import './AllPalettes.scss';
import {deletePalette} from '../../apiCalls'

const AllPalettes = (props) => {
  const {removePalette} = props
  const palettes = props.palettes.map(palette => {
    return(
      <div className='outer-color-container' data-id={palette.id}>
        <h3 className='all-palette-name'>{palette.name}</h3>
        <div className='inner-color-container'>
          <div  id='col-first'  className='all-palettes-colors' style={{backgroundColor: palette.color1}}>
            {palette.color1}
          </div>
          <div className='all-palettes-colors' style={{backgroundColor: palette.color2}}>
            {palette.color2}
          </div>
          <div className='all-palettes-colors' style={{backgroundColor: palette.color3}}>
            {palette.color3}
          </div>
          <div className='all-palettes-colors' style={{backgroundColor: palette.color4}}>
            {palette.color4}
          </div>
          <div id='col-last' className='all-palettes-colors' style={{backgroundColor: palette.color5}}>
            {palette.color5}
          </div>
        </div>
          <button className='delete-palette' onClick={() => removePalette(palette.id)}>Delete</button>
      </div>
    )
  })
  return (
    <div>
      {palettes}
    </div>
  )
}

export default AllPalettes;