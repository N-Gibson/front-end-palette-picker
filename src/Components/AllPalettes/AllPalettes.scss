.all-palette-name {
  font-size: 1.5em;
  border-bottom: 2px solid black;
  display: inline-block;
  margin-top: 0px;
  padding-bottom: 5px;
}

.outer-color-container {
  border: 1px solid lightgrey;
  margin: 20px;
  padding: 10px;
   &:hover {
    box-shadow: 0 4px 8px 0 lightgrey, 0 6px 20px 1px lightgrey;
   }
}

.inner-color-container {
  display: flex;
}

.all-palettes-colors {
  align-items: flex-end;
  display: flex;
  height: 100px;
  justify-content: center;
  padding-bottom: 5px;
  width: 20%;
}

.delete-palette {
  border-radius: 20px;
  font-size: 1em;
  height: 40px;
  width: 200px;
  margin: 20px 20px 0px 20px;
   &:hover {
    color: #fff;
    background-image: linear-gradient(to right, red,orange,yellow,green,blue,indigo,violet);
     cursor: pointer;
   }
}

button:hover {
  color: #fff;
    background-image: linear-gradient(to right, red,orange,yellow,green,blue,indigo,violet);
     cursor: pointer;
}

#col-first {
  border-radius: 10px 0px 0px 10px; 
}

#col-last {
  border-radius: 0px 10px 10px 0px;
}