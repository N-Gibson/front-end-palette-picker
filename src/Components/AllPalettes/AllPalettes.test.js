import React from 'react';
import { shallow } from 'enzyme';
import AllPalettes from './AllPalettes'

describe('All Palettes', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AllPalettes palettes={[{hex:'#333', id:1}, {hex: '#444'}]} />);

  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})