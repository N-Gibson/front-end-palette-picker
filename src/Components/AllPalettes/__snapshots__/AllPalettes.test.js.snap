// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Palettes should match the snapshot 1`] = `
<div>
  <div>
    <h3 />
    <div
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    />
    <div
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    />
    <div
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    />
    <div
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    />
    <div
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    />
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
  <div>
    <h3 />
    <div
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    />
    <div
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    />
    <div
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    />
    <div
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    />
    <div
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    />
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
